### DISCUSSION TITLE
How to get physical path of an image on local file system v4.7.1?

### DISCUSSION THREAD

**Discussion by emclab** [react-native-image-picker/react-native-image-picker#1930] [Q&A]
Besides to uri which is a cache storage of the image just picked, how to get path of the image on local gallery (in both android and iOS)?  I didn't find path in assets object returned. After image is picked, the app needs to know the path of the image when uploading it to cloud. Here is the code for image picker:

```
launchImageLibrary(options, response => {. //options is "photos"
        //console.log("response in pikcupImage : ", response);
        if (response.didCancel) {
          Alert.alert('Cancelled');
          return false;            
        } else if (response.errorCode || response.errorMessage) {
          console.log('User cancelled photo picker');
          return false;
        } else {
          //response.assets is object array
          let temp = [];
          response.assets.forEach((aset, i) => {      
            console.log("aset in picker : ", aset);
            extension = aset.fileName.slice(aset.fileName.lastIndexOf(".") + 1).toLowerCase();
            console.log(" extension ", extension);
            if ((mediaType === 'photo' && imageAllowedExtensions.includes(extension)) || (mediaType === 'video' && videoAllowedExtensions.includes(extension))) {
              aset.path = aset.uri;
              temp.push(aset);
            };                
          }); 
          if (temp && temp.length>0) updatePicks(temp);
          return true;
        }
      })
```

Currently the uploading to cloud only works in android emulator but throws ENOENT (file not found) on real android phone (android 10). Will try to upload with physical path to see if it works.

**Comment by luisresendiz0** [comment]
I'm thinking on mover the caché file into other location to use it

**Comment by skptricks** [comment]
Here is similar example for Display Image From Android Asset Folder In React Native : https://www.skptricks.com/2018/07/display-image-from-android-asset-folder-in-react-native.html
